### ISSUE TITLE
Clp crushes in resize on some big tasks with some options

### ISSUE THREAD

**Issue by svigerske** [coin-or/Clp#40]
Issue created by migration from Trac.

Original creator: alexsaa

Original creation time: 2010-08-18 15:45:17

Assignee: somebody

Version: 

To reproduce:

1. get clp (the problem was noticed on some svn version, then verified on https://projects.coin-or.org/svn/Clp/stable/1.12)

2. prepare artifical test task (please use perl script attached):
```
 ./makeTask.pl >task.lp
```
3. start command-line clp solver with options:
```
 ./clp task.lp presolve off primalS
```
4. get error:
```
*** buffer overflow detected ***: /tmp/coin-Clp/Clp/src/.libs/lt-clp terminated
```
(detailed error message obtained on 64-bit Linux system will be attached)

At the first glance, it seems like buffer is overflown in ClpModel::resize() in
```
char name[9];
sprintf(name, "C%7.7d", iColumn);
```

**Comment by svigerske** [comment]
Attachment [makeTask.pl](https://www.coin-or.org/issue-attachments/Clp/ticket40/makeTask.pl) by alexsaa created at 2010-08-18 15:50:09

Perl script to generate test task (warning: clp will need ~5 Gb RAM to handle that)

**Comment by svigerske** [comment]
Attachment [error.txt](https://www.coin-or.org/issue-attachments/Clp/ticket40/error.txt) by alexsaa created at 2010-08-18 15:54:43

console output

**Comment by svigerske** [comment]
The problem with name too short was fixed by @jjhforrest in 35ba4f7fa7a3 (had been nice if a note had been left here, too).
With that it seems to run fine:
```
$ ./clp ~/tmp/task.lp presolve off primalS
Coin LP version trunk, build Mar 11 2019
command line - /home/stefan/work/coin/Clp-trunk/Clp/src/.libs/lt-clp /home/stefan/tmp/task.lp presolve off primalS 
Coin0009I  CoinLpIO::readLp(): Maximization problem reformulated as minimization
Perturbing problem by 0.001% of 100000 - largest nonzero change 0.10000217 ( 0.10001619%) - largest zero change 0
0  Obj 0 Primal inf 64064690 (6999) Dual inf 3497.3858 (3500)
0  Obj 0 Primal inf 64064690 (6999) Dual inf 6.96401e+13 (6999)
214  Obj 1.6988599e+13 Primal inf 62936034 (6785) Dual inf 6.7504668e+13 (6785)
428  Obj 4.2633198e+13 Primal inf 61738358 (6571) Dual inf 6.5369766e+13 (6571)
642  Obj 7.2947797e+13 Primal inf 60306590 (6357) Dual inf 6.3234664e+13 (6357)
856  Obj 9.1494596e+13 Primal inf 58840743 (6143) Dual inf 6.1099161e+13 (6143)
1070  Obj 1.1121339e+14 Primal inf 57605625 (5929) Dual inf 5.8963559e+13 (5929)
1284  Obj 1.3671739e+14 Primal inf 56209401 (5715) Dual inf 5.6828157e+13 (5715)
1498  Obj 1.4802449e+14 Primal inf 55104961 (5501) Dual inf 5.4691655e+13 (5501)
1712  Obj 1.6757149e+14 Primal inf 53951192 (5287) Dual inf 5.2555953e+13 (5287)
1926  Obj 1.8528279e+14 Primal inf 52668687 (5073) Dual inf 5.0420451e+13 (5073)
2140  Obj 2.1340829e+14 Primal inf 51333401 (4859) Dual inf 4.8285549e+13 (4859)
2354  Obj 2.3472709e+14 Primal inf 49906020 (4645) Dual inf 4.6150446e+13 (4645)
2568  Obj 2.6638709e+14 Primal inf 47678741 (4431) Dual inf 4.4015344e+13 (4431)
2782  Obj 2.8806028e+14 Primal inf 45435621 (4217) Dual inf 4.1879442e+13 (4217)
2996  Obj 3.1235368e+14 Primal inf 43185678 (4003) Dual inf 3.974384e+13 (4003)
3210  Obj 3.3080998e+14 Primal inf 41001786 (3789) Dual inf 3.7608538e+13 (3789)
3424  Obj 3.4776888e+14 Primal inf 38775300 (3575) Dual inf 3.5472936e+13 (3575)
3638  Obj 3.7309438e+14 Primal inf 36570361 (3361) Dual inf 3.3338334e+13 (3361)
3852  Obj 3.9643687e+14 Primal inf 34188699 (3147) Dual inf 3.1202431e+13 (3147)
4066  Obj 4.1528967e+14 Primal inf 31833614 (2933) Dual inf 2.9066829e+13 (2933)
4280  Obj 4.3646967e+14 Primal inf 29426518 (2719) Dual inf 2.6931327e+13 (2719)
4494  Obj 4.9381537e+14 Primal inf 27094186 (2505) Dual inf 2.4799525e+13 (2505)
4708  Obj 7.1317757e+14 Primal inf 24900564 (2291) Dual inf 2.2680923e+13 (2291)
4922  Obj 9.4840157e+14 Primal inf 22548324 (2077) Dual inf 2.0562321e+13 (2077)
5136  Obj 1.1797847e+15 Primal inf 20234493 (1863) Dual inf 1.8443719e+13 (1863)
5350  Obj 1.4149038e+15 Primal inf 17883302 (1649) Dual inf 1.6325116e+13 (1649)
5564  Obj 1.6428448e+15 Primal inf 15603892 (1435) Dual inf 1.4206514e+13 (1435)
5778  Obj 1.8831679e+15 Primal inf 13200661 (1221) Dual inf 1.2087912e+13 (1221)
5992  Obj 2.1175144e+15 Primal inf 10857196 (1007) Dual inf 9.9693101e+12 (1007)
6206  Obj 2.3427171e+15 Primal inf 8605169 (793) Dual inf 7.8507079e+12 (793)
6420  Obj 2.5807304e+15 Primal inf 6225036 (579) Dual inf 5.7321058e+12 (579)
6634  Obj 2.8057206e+15 Primal inf 3975134 (365) Dual inf 3.6135036e+12 (365)
6848  Obj 3.0405191e+15 Primal inf 1627149 (151) Dual inf 1.4949015e+12 (151)
6999  Obj 3.203234e+15
Optimal - objective value 3.203234e+15
Pass 1 took 6999 iterations, objective 3.203234e+15, dual infeasibilities 1.22465e+15( 12246500)
0  Obj 3.203234e+15 Dual inf 1.3998e+12 (13998)
60  Obj 3.1958337e+15
Optimal - objective value 3.1958337e+15
sample size increased from 20997 to 25196
Pass 2 took 60 iterations, objective 3.1958337e+15, dual infeasibilities 1.2225506e+15( 12225506)
6 slacks added
0  Obj 3.1958337e+15 Dual inf 1.8197e+12 (18197)
22  Obj 3.1872011e+15
Optimal - objective value 3.1872011e+15
sample size increased from 25196 to 30235
Pass 3 took 22 iterations, objective 3.1872011e+15, dual infeasibilities 1.2201013e+15( 12201013)
13 slacks added
0  Obj 3.1872011e+15 Dual inf 2.3236e+12 (23236)
20  Obj 3.1773368e+15
Optimal - objective value 3.1773368e+15
sample size increased from 30235 to 36282
Pass 4 took 20 iterations, objective 3.1773368e+15, dual infeasibilities 1.2173021e+15( 12173021)
21 slacks added
0  Obj 3.1773368e+15 Dual inf 2.9283e+12 (29283)
16  Obj 3.165015e+15
Optimal - objective value 3.165015e+15
sample size increased from 36282 to 12253499
Pass 5 took 16 iterations, objective 3.165015e+15, dual infeasibilities 1.2138031e+15( 12138031)
0  Obj 3.165015e+15 Dual inf 1.2138031e+15 (12138031)
214  Obj 3.0199937e+15 Dual inf 1.1255856e+15 (11255856)
428  Obj 2.8702233e+15 Dual inf 1.0587891e+15 (10587891)
Stopped - objective value 2.8702233e+15
row inf 0
column inf 0
sample size increased from 20997 to 23096
Pass 6 took 428 iterations, objective 2.8702233e+15, dual infeasibilities 1.0587891e+15( 10587891)
233 slacks added
0  Obj 2.8702233e+15 Dual inf 1.6097e+12 (16097)
31  Obj 2.8591588e+15
Optimal - objective value 2.8591588e+15
sample size increased from 23096 to 27715
Pass 7 took 31 iterations, objective 2.8591588e+15, dual infeasibilities 1.055883e+15( 10558830)
242 slacks added
0  Obj 2.8591588e+15 Dual inf 2.0716e+12 (20716)
12  Obj 2.850292e+15
Optimal - objective value 2.850292e+15
sample size increased from 27715 to 33258
Pass 8 took 12 iterations, objective 2.850292e+15, dual infeasibilities 1.053626e+15( 10536260)
250 slacks added
0  Obj 2.850292e+15 Dual inf 2.6259e+12 (26259)
29  Obj 2.8377202e+15
Optimal - objective value 2.8377202e+15
sample size increased from 33258 to 39909
Pass 9 took 29 iterations, objective 2.8377202e+15, dual infeasibilities 1.0503981e+15( 10503981)
261 slacks added
0  Obj 2.8377202e+15 Dual inf 3.291e+12 (32910)
31  Obj 2.8217053e+15
Optimal - objective value 2.8217053e+15
sample size increased from 39909 to 12253499
Pass 10 took 31 iterations, objective 2.8217053e+15, dual infeasibilities 1.0465221e+15( 10465221)
0  Obj 2.8217053e+15 Dual inf 1.0465221e+15 (10465221)
214  Obj 2.7044048e+15 Dual inf 9.850092e+14 (9850092)
428  Obj 2.5687881e+15 Dual inf 9.29907e+14 (9299070)
Stopped - objective value 2.5687881e+15
row inf 0
column inf 0
sample size increased from 20997 to 23096
Pass 11 took 428 iterations, objective 2.5687881e+15, dual infeasibilities 9.29907e+14( 9299070)
477 slacks added
0  Obj 2.5687881e+15 Dual inf 1.6097e+12 (16097)
21  Obj 2.5603303e+15
Optimal - objective value 2.5603303e+15
sample size increased from 23096 to 27715
Pass 12 took 21 iterations, objective 2.5603303e+15, dual infeasibilities 9.280952e+14( 9280952)
485 slacks added
0  Obj 2.5603303e+15 Dual inf 2.0716e+12 (20716)
10  Obj 2.5517552e+15
Optimal - objective value 2.5517552e+15
sample size increased from 27715 to 33258
Pass 13 took 10 iterations, objective 2.5517552e+15, dual infeasibilities 9.26282e+14( 9262820)
493 slacks added
0  Obj 2.5517552e+15 Dual inf 2.6259e+12 (26259)
43  Obj 2.5418694e+15
Optimal - objective value 2.5418694e+15
sample size increased from 33258 to 39909
Pass 14 took 43 iterations, objective 2.5418694e+15, dual infeasibilities 9.217271e+14( 9217271)
504 slacks added
0  Obj 2.5418694e+15 Dual inf 3.291e+12 (32910)
45  Obj 2.5251026e+15
Optimal - objective value 2.5251026e+15
sample size increased from 39909 to 12253499
Pass 15 took 45 iterations, objective 2.5251026e+15, dual infeasibilities 9.17785e+14( 9177850)
0  Obj 2.5251026e+15 Dual inf 9.17785e+14 (9177850)
214  Obj 2.3896136e+15 Dual inf 8.5727e+14 (8572700)
428  Obj 2.2664912e+15 Dual inf 8.07279e+14 (8072790)
Stopped - objective value 2.2664912e+15
row inf 0
column inf 0
sample size increased from 20997 to 23096
Pass 16 took 428 iterations, objective 2.2664912e+15, dual infeasibilities 8.07279e+14( 8072790)
734 slacks added
0  Obj 2.2664912e+15 Dual inf 1.6097e+12 (16097)
13  Obj 2.2584168e+15
Optimal - objective value 2.2584168e+15
sample size increased from 23096 to 27715
Pass 17 took 13 iterations, objective 2.2584168e+15, dual infeasibilities 8.058897e+14( 8058897)
741 slacks added
0  Obj 2.2584168e+15 Dual inf 2.0716e+12 (20716)
46  Obj 2.2461315e+15
Optimal - objective value 2.2461315e+15
sample size increased from 27715 to 33258
Pass 18 took 46 iterations, objective 2.2461315e+15, dual infeasibilities 8.03379e+14( 8033790)
753 slacks added
0  Obj 2.2461315e+15 Dual inf 2.6259e+12 (26259)
46  Obj 2.2303951e+15
Optimal - objective value 2.2303951e+15
sample size increased from 33258 to 39909
Pass 19 took 46 iterations, objective 2.2303951e+15, dual infeasibilities 8.002985e+14( 8002985)
767 slacks added
0  Obj 2.2303951e+15 Dual inf 3.291e+12 (32910)
35  Obj 2.211646e+15
Optimal - objective value 2.211646e+15
sample size increased from 39909 to 12253499
Pass 20 took 35 iterations, objective 2.211646e+15, dual infeasibilities 7.969328e+14( 7969328)
0  Obj -34251513 Primal inf 22284016 (2609) Dual inf 8.739821e+16 (9188429)
214  Obj -34166529 Primal inf 21033645 (2454) Dual inf 7.447414e+16 (8571864)
428  Obj -34166529 Primal inf 20013625 (2349) Dual inf 7.0131547e+09 (8089870)
642  Obj -34166529 Primal inf 18921523 (2244) Dual inf 6.6125437e+09 (7657311)
856  Obj -34166529 Primal inf 17904274 (2138) Dual inf 6.233737e+09 (7234060)
1070  Obj -34166529 Primal inf 17010883 (2051) Dual inf 5.8855899e+09 (6856087)
1284  Obj -34166529 Primal inf 16259679 (1957) Dual inf 5.7307401e+09 (6700159)
1498  Obj -34166529 Primal inf 15320680 (1851) Dual inf 5.4394291e+09 (6369115)
1712  Obj -34166529 Primal inf 14508474 (1743) Dual inf 5.18998e+09 (6087770)
1926  Obj -34166529 Primal inf 13514919 (1643) Dual inf 4.9470316e+09 (5805619)
2140  Obj -34166529 Primal inf 12356382 (1544) Dual inf 4.7198807e+09 (5534571)
2354  Obj -34166529 Primal inf 11240436 (1446) Dual inf 4.5174231e+09 (5280394)
2568  Obj -34166529 Primal inf 10442870 (1365) Dual inf 4.3851689e+09 (5114705)
2782  Obj -34166529 Primal inf 9551150 (1283) Dual inf 4.1324245e+09 (4800835)
2996  Obj -34166529 Primal inf 8778143 (1215) Dual inf 4.0041801e+09 (4655855)
3210  Obj -34166529 Primal inf 7955070 (1138) Dual inf 3.8464348e+09 (4455797)
3424  Obj -34166529 Primal inf 7126943 (1051) Dual inf 3.7457739e+09 (4323799)
3638  Obj -34166529 Primal inf 6267261 (969) Dual inf 3.5090776e+09 (4013986)
3852  Obj -34166529 Primal inf 5453573 (885) Dual inf 3.332786e+09 (3784426)
4066  Obj -34166529 Primal inf 4628198 (798) Dual inf 3.0448159e+09 (3427501)
4280  Obj -34166529 Primal inf 3945015 (705) Dual inf 2.7493586e+09 (3031291)
4494  Obj -34165538 Primal inf 3309364 (632) Dual inf 2.4277742e+09 (2627346)
4708  Obj -34165538 Primal inf 2761753 (560) Dual inf 2.122582e+09 (2248156)
4922  Obj -34164597 Primal inf 2165563 (482) Dual inf 1.6207837e+09 (1638775)
5136  Obj -33729666 Primal inf 1697316 (377) Dual inf 1.2893693e+09 (1290660)
5350  Obj -33293072 Primal inf 1260722 (304) Dual inf 1.0568721e+09 (1057930)
5564  Obj -32900160 Primal inf 867810 (218) Dual inf 1.1593275e+09 (1160488)
5778  Obj -32461232 Primal inf 428882 (127) Dual inf 4.9545205e+08 (495948)
5992  Obj -32048357 Primal inf 16007 (4) Dual inf 13970016 (13984)
5999  Obj -32032350
Optimal - objective value -32032350
Optimal objective -32032350 - 14762 iterations time 170.052
```
